src/essreflectometry/reflectometry/types.py

```diff
@@ -0,0 +1,67 @@
+from typing import NewType, TypeVar
+
+import sciline
+import scipp as sc
+
+Reference = NewType('Reference', str)
+Sample = NewType('Sample', str)
+Run = TypeVar('Run', Reference, Sample)
+
+
+class Raw(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Raw data"""
+
+
+class WavelengthData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Raw data transformed to wavelength"""
+
+
+class ThetaData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Wavelength data transformed to theta"""
+
+
+class QData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Theta data transformed to momentum transfer"""
+
+
+QStd = NewType('QStd', sc.Variable)
+
+
+class FootprintCorrected(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Experiment data corrected by footprint on sample"""
+
+
+WavelengthResolution = NewType('WavelengthResolution', sc.Variable)
```

Comment:
It's nice that docs strings of `Scope` types (such as [esssans.types.CalibratedMaskedData](https://scipp.github.io/esssans/generated/modules/esssans.types.CalibratedMaskedData.html)) are displayed in the API-reference.
It would be good to show the docs strings explaining the other non-scope types (such as [essans.types.TransmissionFraction](https://scipp.github.io/esssans/generated/modules/esssans.types.TransmissionFraction.html)) in the documentation as well.

Is there some way we can achieve that without making all provider types into classes?
Otherwise perhaps we should consider making all provider types into classes just to make the docs strings display in the documentation.

Reply:
Not sure I understand the question. Just add a docstring to the `NewType`, it works fine in `esssans`: https://scipp.github.io/esssans/generated/modules/esssans.types.html#module-esssans.types

Reply:
I misunderstood. I'll add docstrings.